Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -183,12 +183,12 @@ class SyncRepository {
         /// 1. Collect all the **active** channel ids
         operations.append(ActiveChannelIdsOperation(syncRepository: self, context: context))
         
-        // 2. /sync
-        operations.append(SyncEventsOperation(syncRepository: self, context: context, recovery: false))
-        
-        // 3. Refresh channel lists (required even after applying events)
+        // 2. Refresh channel lists
         operations.append(contentsOf: activeChannelLists.allObjects.map { RefreshChannelListOperation(channelList: $0, context: context) })
         operations.append(contentsOf: activeChannelListControllers.allObjects.map { RefreshChannelListOperation(controller: $0, context: context) })
+
+        // 3. /sync (for channels what not part of active channel lists)
+        operations.append(SyncEventsOperation(syncRepository: self, context: context, recovery: false))
```

Comment:
Why would we want to sync channels that are not part of active channel lists? 🤔 I think we probably could discard these events. These channels will have the most up-to-date data whenever we refetch them

Reply:
One case I could think of is channel controller targeting a channel id which has not been loaded by the channel list yet (but sure yes it sounds like a special case). Or some sort of custom integrations not having channel list alive (e.g. operating on channel ids). Hard to know how valid these cases are.

Reply:
There are tab/segmented control based channel lists, which are not active at all times, but can be invoked with a tap. So I agree with Toomas on this one to keep this logic.

Reply:
@martinmitrevski @laevandus If they are not active, meaning, they have not be loaded yet, when a user opens a tab, we will fetch the data from the server, the most up-to-date one, so the events are useless

Reply:
I was thinking about the case where one creates channel controller, then app goes to background for awhile, comes back, then sync repo fetches events what the user missed.

Reply:
But in that case, the controller should be part of the active controllers I believe

Reply:
We do not call update each on each active channel controller (relies on /sync, otherwise too many API calls)